csrc/iter_visitor.cpp

```diff
@@ -227,6 +228,23 @@ void IterVisitor::traverseBetween(
         // If we don't want to retraverse, remove nodes we already visisted.
         remove_visited(next_stmts, visited);
       }
+
+      if (traverse_siblings) {
+        // Add unvisited siblings to next_stmts
+        std::vector<Statement*> unvisited_sibs;
+        for (auto next_val : ir_utils::filterByType<Val>(next_stmts)) {
+          for (auto sib : ir_utils::siblingValsOf(next_val)) {
+            if (visited.find(sib) == visited.end()) {
+              // Push to separate vector so that we don't modify next_stmts
+              // while looping
+              unvisited_sibs.push_back(sib);
+            }
+          }
+        }
+        next_stmts.insert(
+            next_stmts.end(), unvisited_sibs.begin(), unvisited_sibs.end());
+      }
```

Comment:
A couple of questions.

- Do we want to visit siblings even when they are in `visited` if `traverse_all_paths` is true?
- More commonly, though, since when we are visiting an input first, if that has a sibling, it would be most likely not visited yet, no matter if it's an orphaned sibling or not. Wouldn't this change the traversal ordering then? It would still be a topological order, but I'd be surprised if `traverse_siblings` would change the traversal order of a fusion with no orphaned siblings.
- Can you please add unit tests just for this traversal with siblings?

Reply:
1. Yeah, we should add them even if visited here I believe.
2. You're right that this flag changes the ordering. An alternative that would not change the ordering would be to add siblings to a vector called `maybe_orphaned_siblings` here, then after the `while (!stmt_stack.empty()) {}` loop we would loop over `maybe_orphaned_siblings` and call handle on those that are not found in `visited`. This would use some extra space to hold the vector of siblings, but would perform the same number of searches in `visited`.
3. Yes. I will add some tests for the generic traversal.

Reply:
Re 2: The alternative sounds simpler and better to me. The extra space would be negligible.

Reply:
Pushed changes that address 1 and 2 above. Will add tests soon

Reply:
Pushed tests, and fixed a bug causing siblings of outputs to not be traversed.